R/grid_code_paths.R

```diff
@@ -5,13 +5,9 @@ tune_grid_loop <- function(resamples, grid, workflow, metrics, control, rng) {
   packages <- c(control$pkgs, required_pkgs(workflow))
 
 
-  is_h2o <- is_h2o(workflow)
   # TODO: remove when h2o supports non-regular grid
-  regular_grid <- is.null(grid) || is_regular_grid(grid)
-  if (is_h2o) {
-    if (!regular_grid) {
-      rlang::abort("The `h2o` engine only supports regular tuning grid.")
-    }
+  if (!is.null(grid) && is_h2o(workflow) && !is_regular_grid(grid)) {
```

Comment:
h2o currently only supports regular grid. If the grid is null then we are in `fit_resamples`. no need to check.